src/index.ts

```diff
@@ -185,17 +178,15 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
       request.headers.has('if-none-match'),
       response.headers.has('etag'),
       request.headers.get('if-none-match') === `${pathKey}`,
-    ].every((val) => val === true)
+    ].every(Boolean)
 
     if (shouldRevalidate) {
       // TypeError permitted to satisfy mocks per /issues/96
-      try {
-        // Proactively remove body per /pull/94#discussion_r425455176
-        response.body.cancel()
-      } catch (e) {
-        if (e instanceof TypeError && e.message === 'response.body.cancel is not a function') {
-          console.log('Environment doesnt support readable streams')
-        }
+      const cancelResponse = response.body.cancel;
```

Comment:
In testing, I realized `response.body` can be null, because `response` may actually be a `304 Not Modified` response, which does not have a body. In that case, this will throw a `TypeError`.

The call to `cache.match(request)` returns a `304  Not Modified` response if a match is found with an ETag header with a value that matches a value in the request's `If-None-Match` header. More here:  https://developers.cloudflare.com/workers/reference/apis/cache/